app/src/main/java/com/aptoide/uploader/account/AccountFactory.java

```
@@ -0,0 +1,17 @@
+package com.aptoide.uploader.account;
+
+public class AccountFactory {
+
+  public static Account of(boolean hasStore, boolean isLoggedIn, String getStoreName,
+      BaseAccount.LoginType loginType) {
+    switch (loginType) {
+      case APTOIDE:
+        return new AptoideAccount(hasStore, isLoggedIn, getStoreName);
+      case GOOGLE:
+        return new GoogleAccount(hasStore, isLoggedIn, getStoreName);
+      case FACEBOOK:
+        return new FacebookAccount(hasStore, isLoggedIn, getStoreName);
+    }
+    return new AptoideAccount(hasStore, isLoggedIn, getStoreName);
```

I prefer throwing a runtime exception rather than having a default that shouldn't happen. It might bring us problems in the future.